stl/inc/random

```diff
@@ -4507,7 +4507,7 @@ private:
     result_type _Eval(_Engine& _Eng, const param_type& _Par0) const {
         double _Px           = _NRAND(_Eng, double);
         const auto _First    = _Par0._Pcdf.begin();
-        const auto _Position = _STD lower_bound(_First, _Par0._Pcdf.end(), _Px);
+        const auto _Position = _STD lower_bound(_First, _STD prev(_Par0._Pcdf.end()), _Px);

```

Comment:
We have a utility function `_Prev_iter(i)` that's equivalent to `std::prev(i, 1)` except that it uses a single decrement instead of subtracting `1` since decrement is sometimes cheaper for complicated iterator types:
```suggestion
        const auto _Position = _STD lower_bound(_First, _Prev_iter(_Par0._Pcdf.end()), _Px);
```